### ISSUE TITLE
How to update the cached image if the image in server changed or deleted . I have used Library using Pod and in Objective c 

### ISSUE THREAD

**Issue by Srijeeth** [SDWebImage/SDWebImage#2812]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dreampiggy** [comment]
Use [SDWebImageRefreshCached](https://sdwebimage.github.io/Enums/SDWebImageOptions.html#/c:@E@SDWebImageOptions@SDWebImageRefreshCached) option may help with this use case.

**Comment by Srijeeth** [comment]
Thanks for the reply @dreampiggy . 
It dint Work @dreampiggy  Is there any way around. i have used code i this way
 [cell.itemImageView sd_setImageWithURL:[NSURL URLWithString:path]
                                      placeholderImage:[UIImage imageNamed:@"nopreview.jpg"] options:SDWebImageRefreshCached
                                             completed: nil];

**Comment by Srijeeth** [comment]
Thanks for the reply @dreampiggy .
It dint Work @dreampiggy Is there any way around. i have used code i this way
[cell.itemImageView sd_setImageWithURL:[NSURL URLWithString:path]
placeholderImage:[UIImage imageNamed:@"nopreview.jpg"] options:SDWebImageRefreshCached
completed: nil];

**Comment by Srijeeth** [comment]
I tried using refreshcached but it din work

**Comment by Srijeeth** [comment]
Hello sir,
Is there any solution.i tried using Refreshimagecached it dint work..  Need
a solution asap please.

**Comment by kinarobin** [comment]
It didn't work with `SDWebImageRefreshCached` ? Can you describe this issue in more detail?

**Comment by Srijeeth** [comment]
It dint work. Once the image loaded from server in tableviewcell  at that
moment if I delete the image from server that image still exists in cache.

**Comment by kinarobin** [comment]
Oh, Once you download image , the image will be cached in memory.
You should reload the tableView, let the image reload from the server.

**Comment by Srijeeth** [comment]
I did  but it's not happening Bro can u help me in this

**Comment by kinarobin** [comment]
Can you upload a demo? that I can debug with the issue.

**Comment by Srijeeth** [comment]
*  [cell.itemImageView sd_setImageWithURL:[NSURL URLWithString:path]*

*                                      placeholderImage:[UIImage
imageNamed:@"nopreview.jpg"]options:SDWebImageRefreshCached*

*                                             completed: nil];*


*im just uisng this code images ll get downloaded but if we delete the
image from server it will not update here while refreshing *

**Comment by dreampiggy** [comment]
Delete does not means "Cache Miss"

You can change your remote resource into another one but keep the same URL. Then have a test for this.

**Comment by Srijeeth** [comment]
But that will not work sir. I have item browse which loads and my image
name will be item No for eample 12345.jpg in the server

**Comment by dreampiggy** [comment]
@Srijeeth The design of this `RefreshCached` is to solve the problem of `URL same but resource changed, not the URL Same but resource been deleted`.

For this case, if you want to clean your cache, currentlly, you can use the notification to check the server HTTP response code. And then delete the cache. See about documentation:

+ [SDWebImageDownloadReceiveResponseNotification](https://sdwebimage.github.io/Constants.html#/c:@SDWebImageDownloadReceiveResponseNotification)
+ [SDWebImageDownloaderOperation.response](https://sdwebimage.github.io/Classes/SDWebImageDownloaderOperation.html#/c:objc(cs)SDWebImageDownloaderOperation(py)response)

**Comment by Srijeeth** [comment]
Thanks for the reply sir can u tel me how to use it i the code in objective
c

**Comment by bpoplauschi** [comment]
Seems like #2813 is a duplicate. Same user same problem
